src/foam/nanos/notification/push/WebPushService.js

```diff
@@ -115,14 +116,22 @@ foam.CLASS({
       System.err.println("        auth: " + sub.getAuth());
       */
         try {
-          Notification n = new Notification(
-            sub.getEndpoint(),
-            sub.getKey(),  // sub.getUserPublicKey(),
-            sub.getAuth(), // sub.getAuthAsBytes(),
-            msg
-          );
+          if ( sub instanceof foam.nanos.notification.push.iOSNativePushRegistration ) {
```

Comment:
This service and the apple version really aren't decoupled because of this check so I would just combine the two..

Reply:
Yes, but the APNS service does a lot, including finding creds and stuff, so probably better to be separate files

Reply:
But neither part is reusable or replaceable without the other, so they're really just the same feature divided between two files.